### PR TITLE
Visualization perf #45

### PR DESCRIPTION
**Pull request by andersnylund**

### CODE REVIEW

File: src/reducers/amountReducer.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+export const initialState = {
+  amountByParties: [],
+  amountByCategories: []
+};
+
+const reducer = (state = initialState, action) => {
+  if (action.type === 'SET_AMOUNT_BY_PARTY') {
```
Comment (tu-tran) [inline src/reducers/amountReducer.js]: Should this be some "const integer" instead of the whole stringy ?

Comment (andersnylund) [reply src/reducers/amountReducer.js]: True that. Forgot to change it here

**Comment by tu-tran** [comment]
Hmm, is the purpose of this change is so that all calculation will  be cached? Apparently when I revisit the "Category" tab in Visualization... There is still a delay. And shall we consider using some kind of virtualized data grid in the Transaction List too? (at least it takes a good few seconds to display my fake transactions) :(

**Comment by andersnylund** [comment]
The idea was to have a re-calculation triggered each time a new categorization happens so that the calculation is not done in the component that does the visualization e.g. `CategoryGrouping.js`. The result with the previous approach was that when you clicked from that to tab, everything was calculated again (and that took a looong time).

I think that the problem when opening categorization page is a different problem. I went back in git history to commit `cbd3775` where we didn't have redux yet. I used the same large amount of transactions, and the categorization page was already slow there. Therefore I think that the slowness of categorization page is due to react's slow rendering performance, and is not affected that much by the calculation.

There is a lot to improve with performance, and I think that the first step would be to learn how to overall write performant code in React. I have myself no knowledge of that at this point 😅 https://medium.com/@joomiguelcunha/react-performance-tips-5fa199a450b2 could be a good start

**Pull request merged**